src/lib/storage/planned_volumes_collection.rb

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "storage/planned_volume"
+
+module Yast
+  module Storage
+    # Set of PlannedVolume elements
+    #
+    # Implements Enumerable and provides some extra methods to query the set
+    class PlannedVolumesCollection
```

Comment:
Good idea to move this out to a separate class. I had toyed with that idea, but not really done it for lack of extra functionality, but those few extra methods here really make it worthwhile.